src/modm/driver/display/ssd1306_i2c.lb

```diff
@@ -12,19 +12,20 @@
 
 
 def init(module):
-    module.name = ":driver:ssd1306"
-    module.description = "SSD1306 Display"
+    module.name = ":driver:ssd1306.i2c"
```

Comment:
Should we add an alias here?
If so, how does it work?

Reply:
yes, with `module.add_alias` in the `:driver` module. [See example here](https://github.com/modm-io/modm/blob/develop/tools/build_script_generator/scons/module.lb#L43-L59). Also add a DEPRECATED: 2024q4 so we can remove it and not forget it like in my own example \*cough\*😇.

Reply:
I added the following code to `src/modm/driver/module.lb`:
```python
# DEPRECATED: 2024q4
module.add_alias(
    Alias(name="ssd1306",
          destination=":ssd1306.i2c",
          description="DEPRECATED: Please use :driver:ssd1306.i2c instead."))
```
lbuild recognizes the alias and does not complain if I use it in `project.xml`, but does not generate any files...

Reply:
needs to be `destination=":driver:ssd1306.i2c",` I'll check why it doesn't complain.

Reply:
Nope, still does not work 😞

Reply:
Oh ffs, I'll check it over the weekend. I need to rewrite lbuild, it's so unpythonic…

Reply:
It does build the alias destination for me though.

The reason it doesn't give a warning is that the `:driver:ssd1306.i2c` module is a dependency of the `:driver:sh1106.i2c` module, and is thus already included by the time the alias is resolved. If you change the dependency back to the alias:

```
 $ lbuild build
[WARNING] lbuild.node: Node 'modm:driver:ssd1306' has been moved to ':driver:ssd1306.i2c'!

>> modm:driver:ssd1306  [Alias]

DEPRECATED: Please use `:driver:ssd1306.i2c` instead.

Destination: :driver:ssd1306.i2c

>> modm:driver:ssd1306.i2c  [Module]

SSD1306 Display in I2C mode
```